test/shared/PosmTestSetup.sol

```diff
@@ -67,13 +71,19 @@ contract PosmTestSetup is Test, Deployers, DeployPermit2, LiquidityOperations {
     function deployPosm(IPoolManager poolManager) internal {
         // We use deployPermit2() to prevent having to use via-ir in this repository.
         permit2 = IAllowanceTransfer(deployPermit2());
-        positionDescriptor =
-            Deploy.positionDescriptor(address(poolManager), 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2, "ETH", hex"00");
+        proxyAsImplementation = deployDescriptor(poolManager, "ETH");
         lpm = Deploy.positionManager(
-            address(poolManager), address(permit2), 100_000, address(positionDescriptor), address(_WETH9), hex"03"
+            address(poolManager), address(permit2), 100_000, address(proxyAsImplementation), address(_WETH9), hex"03"
         );
     }
 
+    function deployDescriptor(IPoolManager poolManager, bytes32 label) internal returns (IPositionDescriptor) {
+        positionDescriptor =
+            Deploy.positionDescriptor(address(poolManager), 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2, label, hex"00");
```

Comment:
why is that address hardcoded for native? shouldn't it be address(_WETH9)?

Reply:
its hardcoded so i can check if its equal in PositionDescriptor.t.sol
if it wasnt hardcoded, foundry would deploy WETH9 to a random address

Reply:
could do it the opposite way

Reply:
its just that if we want to run fork tests this would break no?

Reply:
hm i guess not if we're not doing any actual transfers with that token, it is odd though that the weth we pass into posm will be different than the weth in position descriptor